x/ecocredit/core/msg_update_project_metadata.go

```diff
@@ -0,0 +1,41 @@
+package core
+
+import (
+	"github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/x/auth/legacy/legacytx"
+
+	"github.com/regen-network/regen-ledger/x/ecocredit"
+)
+
+var _ legacytx.LegacyMsg = &MsgUpdateProjectMetadata{}
+
+func (m MsgUpdateProjectMetadata) ValidateBasic() error {
+	if _, err := types.AccAddressFromBech32(m.Admin); err != nil {
+		return sdkerrors.ErrInvalidAddress.Wrap(err.Error())
+	}
+	metadataLen := len(m.NewMetadata)
+	if metadataLen == 0 {
+		return sdkerrors.ErrInvalidRequest.Wrap("metadata cannot be empty")
+	}
```

Comment:
Metadata is currently any arbitrary metadata, including empty metadata. We currently don't check for empty metadata when creating or updating a credit class, when creating a project, or when creating a credit batch. Not sure if there is a use case for removing metadata from a project but I think we should be consistent for now and maybe further discuss on our product call.

Reply:
https://github.com/regen-network/regen-ledger/pull/1009/commits/0f798b40e62794309e98ed0a5a35658a67519404

fixed